pkg/sentry/kernel/semaphore/semaphore.go

```diff
@@ -337,19 +337,16 @@ func (s *Set) Size() int {
 	return len(s.sems)
 }
 
-// Change changes some fields from the set atomically.
-func (s *Set) Change(ctx context.Context, creds *auth.Credentials, owner fs.FileOwner, perms fs.FilePermissions) error {
+// Set modifies attributes for a semaphore set. See semctl(IPC_SET).
+func (s *Set) Set(ctx context.Context, ds *linux.SemidDS) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	// "The effective UID of the calling process must match the owner or creator
-	// of the semaphore set, or the caller must be privileged."
-	if !s.obj.CheckOwnership(creds) {
-		return linuxerr.EACCES
+	err := s.obj.Set(ctx, &ds.SemPerm)
```

Comment:
Does this now change the mode? To preserve the mode, you might need to update `ds.SemPerm.Mode`.

Reply:
Yes, it does change it. The man page for `semctl(2)` mentions under `IPC_SET`: 
```
    The following members of the structure are updated:
    sem_perm.uid, sem_perm.gid, and (the least significant 9
    bits of) sem_perm.mode.
```
It's either an error in the original, or maybe I'm missing something.

Reply:
nit:

if err := .... ; err != nil {
  ...
}

Reply:
Done